recipes/sdl_ttf/all/conanfile.py

```diff
@@ -121,6 +123,10 @@ def package_info(self):
         self.cpp_info.components["_sdl2_ttf"].requires = ["freetype::freetype", "sdl::libsdl2"]
         if self.options.get_safe("with_harfbuzz"):
             self.cpp_info.components["_sdl2_ttf"].requires.append("harfbuzz::harfbuzz")
+        if Version(self.version) <= "2.0.18" and is_apple_os(self) and self.options.shared:
+            self.cpp_info.components["_sdl2_ttf"].frameworks = [
+                "AppKit", "CoreGraphics", "CoreFoundation", "CoreServices"
+        ]
```

Comment:
It makes no sense, it should be for static only, not shared.

Reply:
Please open an issue with more information :) If the hooks are incorrect https://github.com/conan-io/conan-center-index/pull/18366#issuecomment-1625310577 please open an issue there so we can track it 🙏

Reply:
Hook is correct, but your interpretation of its result is incorrect.

Reply:
This hook just says that shared lib links to some system libs or frameworks, so it means that these system libs/frameworks must be added if static. This hook has always been an indirect way to find system libs or frameworks to add if static.

Reply:
I'd argue the hooks are bad for providing misleading information

> Library './lib/libSDL2_ttf.dylib' links to system library 

implies the shared one needs it, if that incorrect -- then the messages are unclear -- it should have included the settings and options to be more concise :)

Reply:
Feel free to update message hook if you think it's misleading.

Instead of:

```
post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library './lib/libSDL2_ttf.dylib' links to system library 'CoreServices' but it is not in cpp_info.frameworks.
```

it could be:

```
post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library './lib/libSDL2_ttf.dylib' links to system library 'CoreServices', therefore it should be in cpp_info.frameworks if static, but it's missing.
```

Reply:
Done, thanks for the suggestion!